### DISCUSSION TITLE
How to use the pion ice-lite

### DISCUSSION THREAD

**Discussion by xianchuhan** [pion/webrtc#1983] [Q&A] [answered]
pion webrtcV2 server deploy on the public network cloud host,using an example:
       settingEngine := webrtc.SettingEngine{}
	settingEngine.SetLite(true)
	settingEngine.SetInterfaceFilter(func(ifs string) bool {
		if strings.Contains(ifs, "docker") {
			return false
		}
		return true
	})
	
	
	settingEngine.SetNAT1To1IPs([]string{cf.publicIPs}, webrtc.ICECandidateTypeHost)
        config := webrtc.Configuration{
		        //ICEServers: iceServer[0:],
	        }
	peerConn, err := api.NewPeerConnection(config)

However, pion webrtc found no response from the server - side client stun binding request.

client:
![image](https://user-images.githubusercontent.com/82028984/137280382-b34ac4d2-4ad8-4cf2-b3ce-eadd14a88e03.png)
server:
![image](https://user-images.githubusercontent.com/82028984/137280537-15ee4af1-a564-4f8f-a19f-ef8e2c409069.png)

**Comment by Sean-Der** [comment, marked as answer]
Hi @xianchuhan 

Sorry for the delay. If you don't enable `ice-lite` does Pion work?

**Comment by xianchuhan** [comment]
It doesn't matter. I have found the cause of the problem,because a previous colleague did not use ice-lite mode,pion ICE does not support relay connection first and srflx connection later,a browser role is ice-controlling, it can switch relay to srflx，but pion ice does not support switch，cause not charge each other on both sides of the flow,to resolve this problem,pion webrtc ice-role wrote die for controlling.
Now to ice-lite mode, pion webrtc ice-role not dead.

**Reply by Sean-Der** [reply]
If you want to change candidate pairs after connected you can do an ICE Restart. Does the controlling Agent automatically switch?

It sounds like you might be describing [NICER](https://www.ietf.org/id/draft-oreland-dispatch-ice-nicer-00.html) ?

**Comment by xianchuhan** [comment]
Thanks for your answer.  I thought the rfc document describes,ice-role is controlling can decide whether you can switch link.
